heatmap_printer.go

```diff
@@ -0,0 +1,807 @@
+
+	minValue float32
+	maxValue float32
+
+	rgbLegendValue int
+}
+
+var colorComplement = map[Color]Color{
+	BgBlack:        FgLightWhite,
+	BgRed:          FgCyan,
+	BgGreen:        FgMagenta,
+	BgYellow:       FgBlue,
+	BgBlue:         FgYellow,
+	BgMagenta:      FgGreen,
+	BgCyan:         FgRed,
+	BgWhite:        FgBlack,
+	BgDefault:      FgBlack,
+	BgDarkGray:     FgLightWhite,
+	BgLightRed:     FgLightCyan,
+	BgLightGreen:   FgLightMagenta,
+	BgLightYellow:  FgLightBlue,
+	BgLightBlue:    FgLightYellow,
+	BgLightMagenta: FgLightGreen,
+	BgLightCyan:    FgLightRed,
+	BgLightWhite:   FgBlack,
+}
+
+// WithAxisData returns a new HeatmapPrinter, where the first line and row are headers.
+func (p HeatmapPrinter) WithAxisData(hd HeatmapAxis) *HeatmapPrinter {
+	p.HasHeader = true
+	p.Axis = hd
+	return &p
+}
+
+// WithAxisStyle returns a new HeatmapPrinter with a specific AxisStyle.
+func (p HeatmapPrinter) WithAxisStyle(style *Style) *HeatmapPrinter {
+	p.AxisStyle = style
+	return &p
+}
+
+// WithSeparatorStyle returns a new HeatmapPrinter with a specific SeparatorStyle.
+func (p HeatmapPrinter) WithSeparatorStyle(style *Style) *HeatmapPrinter {
+	p.SeparatorStyle = style
+	return &p
+}
+
+// WithData returns a new HeatmapPrinter with specific Data.
+func (p HeatmapPrinter) WithData(data [][]float32) *HeatmapPrinter {
+	p.Data = data
+	return &p
+}
+
+// WithTextColor returns a new HeatmapPrinter with a specific TextColor.
+func (p HeatmapPrinter) WithTextColor(color Color) *HeatmapPrinter {
+	p.TextColor = color
+	p.EnableRGB = false
+	p.ColorValueComplementarily = false
+	return &p
+}
+
+// WithTextRGB returns a new HeatmapPrinter with a specific TextRGB.
+func (p HeatmapPrinter) WithTextRGB(rgb RGB) *HeatmapPrinter {
+	p.TextRGB = rgb
+	p.EnableRGB = true
+	p.ColorValueComplementarily = false
+	return &p
+}
+
+// WithBoxed returns a new HeatmapPrinter with a box around the table.
+func (p HeatmapPrinter) WithBoxed(b ...bool) *HeatmapPrinter {
+	p.Boxed = internal.WithBoolean(b)
+	if p.Boxed && !p.Grid {
+		p.Grid = true
+	}
+	return &p
+}
+
+// WithGrid returns a new HeatmapPrinter with a grid.
+func (p HeatmapPrinter) WithGrid(b ...bool) *HeatmapPrinter {
+	b2 := internal.WithBoolean(b)
+	p.Grid = b2
+	if !b2 && p.Boxed {
+		p.Boxed = false
+	}
+	return &p
+}
+
+// WithEnableRGB returns a new HeatmapPrinter with RGB colors.
+func (p HeatmapPrinter) WithEnableRGB(b ...bool) *HeatmapPrinter {
+	p.EnableRGB = internal.WithBoolean(b)
+	return &p
+}
+
+// WithOnlyColoredCells returns a new HeatmapPrinter with only colored cells.
+func (p HeatmapPrinter) WithOnlyColoredCells(b ...bool) *HeatmapPrinter {
+	b2 := internal.WithBoolean(b)
+	p.OnlyColoredCells = b2
+	return &p
+}
+
+// WithLegendOnlyColoredCells returns a new HeatmapPrinter with legend with only colored cells.
+func (p HeatmapPrinter) WithLegendOnlyColoredCells(b ...bool) *HeatmapPrinter {
+	b2 := internal.WithBoolean(b)
+	p.LegendOnlyColoredCells = b2
+	if b2 {
+		p.Legend = true
```

Comment:
Please include this behavior in the doc comment.

Reply:
done